src/Workspaces/Core/Desktop/Execution/DesktopReferenceSerializationServiceFactory.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Execution.Serialization;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// this is desktop implementation of IReferenceSerializationService
+    /// </summary>
+    [ExportWorkspaceServiceFactory(typeof(IReferenceSerializationService), layer: ServiceLayer.Desktop), Shared]
+    internal class ReferenceSerializationServiceFactory : IWorkspaceServiceFactory
+    {
+        private static readonly SerializationAnalyzerAssemblyLoader s_loader = new SerializationAnalyzerAssemblyLoader();
+
+        public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
+        {
+            return new Service(workspaceServices.GetService<ITemporaryStorageService>());
+        }
+
+        private sealed class Service : AbstractReferenceSerializationService
```

Comment:
In the future could you provide better names?  "Service" makes things hard to find with features like NavigateTo because it's too vague and generic.